### PR TITLE
[stable/grafana] enable UNIQUE_FILENAMES variable options for kiwigrid/k8s-sidecar

### LINKED ISSUE #23397: [stable/grafana] Enable the UNIQUE_FILENAMES option for the kiwigrid/k8s-sidecar
**Is your feature request related to a problem? Please describe.**
Importing Grafana dashboards from multiple teams sometimes creates filename collisions in the dashboard import directory. This in turn creates race conditions between the dashboards and you only end up with the most recently updated dashboard available in Grafana.

**Describe the solution you'd like**
The kiwigrid/k8s-sidecar project already supports a UNIQUE_FILENAMES environment variable which would fix this situation. It only needs to be exposed for use in the Helm chart.

**Describe alternatives you've considered**
I have not considered any alternatives. I am only raising this so I can reference it from my pull request that will address the issue.

**Additional context**

### PR DESCRIPTION
**Pull request by spykermj**
#### What this PR does / why we need it:

@zanhsieh, @rtluckie, @maorfr 

As detailed in #23397, if many teams are creating Grafana dashboards it is quite easy to have filename collisions in the import directory of grafana. This allows setting the UNIQUE_FILENAMES environment variable in kiwigrid/k8s-sidecar to address the problem.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #23397 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by spykermj** [comment]
/assign @zanhsieh

I believe this is ready to go. Please let me know if I need to do anything more.

File: stable/grafana/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: grafana
-version: 5.5.1
+version: 5.5.2
```
Comment (zanhsieh) [inline stable/grafana/Chart.yaml]: Please bump the chart version to latest. Thank you.

Comment (spykermj) [reply stable/grafana/Chart.yaml]: @zanhsieh, I've now bumped the version to 5.5.3

Comment (spykermj) [reply stable/grafana/Chart.yaml]: @zanhsieh, did you literally mean "latest" ? I can do that if it is what you meant.

Comment (zanhsieh) [reply stable/grafana/Chart.yaml]: Nay. We should stick on specific version number, not latest. I don't think git or helm will take 'latest'.

**Comment by zanhsieh** [comment]
@spykermj 
Could you also review why below stable/grafana/Chart.yaml still conflict?

**Comment by spykermj** [comment]
> @spykermj
> Could you also review why below stable/grafana/Chart.yaml still conflict?

@zanhsieh, I've resolved the conflict.

**Review by zanhsieh — APPROVED** [review]

**Comment by spykermj** [comment]
/retest

I'm struggling to understand the failure reason, maybe it was transient?

**Comment by spykermj** [comment]
@zanhsieh, I'm struggling to understand why the pull-charts-e2e job is failing.

This section from the log seems to be most informative, but I cannot see an actual reason for the failure.


```
I0801 17:26:58.546] E0801 17:26:58.291470     437 portforward.go:372] error copying from remote stream to local connection: readfrom tcp4 127.0.0.1:37585->127.0.0.1:40996: write tcp4 127.0.0.1:37585->127.0.0.1:40996: write: broken pipe
I0801 17:26:59.162] deployment "grafana-7mzmkvlm28" successfully rolled out
I0801 17:26:59.915] RUNNING: grafana-7mzmkvlm28-test
I0801 17:27:02.909] FAILED: grafana-7mzmkvlm28-test, run `kubectl logs grafana-7mzmkvlm28-test --namespace grafana-presubmit-23399-1289613144506765312-7mzmkvlm28` for more info
I0801 17:27:02.948] Error: 1 test(s) failed
I0801 17:27:02.949] ========================================================================================================================
I0801 17:27:02.949] ........................................................................................................................
I0801 17:27:02.949] ==> Events of namespace grafana-presubmit-23399-1289613144506765312-7mzmkvlm28
```

Any ideas what I need to do to get it to pass?

**Comment by spykermj** [comment]
@zanhsieh, I have confirmed that the "Test Health" test failure I'm getting is not actually introduced by my change. I reset my branch to upstream/master, then only bumped the chart version and "Test Health" failed in the pull-charts-e2e job.

How can I get the pull-charts-e2e to pass?

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**